tests/kafkatest/tests/core/separate_interbroker_listener_test.py

```diff
@@ -0,0 +1,105 @@
+
+    def setUp(self):
+        self.topic = 'test_topic'
+        self.zk = ZookeeperService(self.test_context, num_nodes=1)
+        self.kafka = KafkaService(self.test_context, num_nodes=3, zk=self.zk, topics={self.topic: {
+            "partitions": 3,
+            "replication-factor": 3,
+            'configs': {"min.insync.replicas": 2}}})
+        self.zk.start()
+
+    def create_producer_and_consumer(self):
+        self.producer = VerifiableProducer(
+            self.test_context, 1, self.kafka, self.topic, throughput=100)
+
+        self.consumer = ConsoleConsumer(
+            self.test_context, 1, self.kafka, self.topic, consumer_timeout_ms=60000, message_validator=is_int)
+
+    def roll_in_interbroker_listener(self, broker_protocol, broker_sasl_mechanism, use_separate_listener=False):
+        self.kafka.setup_interbroker_listener(broker_protocol, use_separate_listener)
+        self.kafka.interbroker_sasl_mechanism = broker_sasl_mechanism
+        self.bounce()
+
+    def bounce(self):
+        self.kafka.start_minikdc()
+        for node in self.kafka.nodes:
+            self.kafka.stop_node(node)
+            self.kafka.start_node(node)
+            time.sleep(10)
+
+    @cluster(num_nodes=9)
+    @parametrize(client_protocol=SecurityConfig.SASL_SSL, broker_protocol=SecurityConfig.SASL_SSL,
+                 broker_sasl_mechanism=SecurityConfig.SASL_MECHANISM_GSSAPI)
+    @parametrize(client_protocol=SecurityConfig.SASL_SSL, broker_protocol=SecurityConfig.SASL_SSL,
+                 broker_sasl_mechanism=SecurityConfig.SASL_MECHANISM_PLAIN)
+    @parametrize(client_protocol=SecurityConfig.SSL, broker_protocol=SecurityConfig.SASL_SSL,
+                 broker_sasl_mechanism=SecurityConfig.SASL_MECHANISM_GSSAPI)
+    def test_enable_separate_interbroker_listener(self, client_protocol, broker_protocol, broker_sasl_mechanism):
+        """
+        Start with a cluster that has a single {{client_protocol}} listener.
+        Start producer and consumer on the {{client_protocol}} listener.
+        Open a SECURED dedicated interbroker port via rolling upgrade.
+        Ensure we can produce and consume via {{client_protocol}} port throughout.
+        """
+        client_sasl_mechanism = SecurityConfig.SASL_MECHANISM_GSSAPI
+
+        self.kafka.security_protocol = client_protocol
+        self.kafka.client_sasl_mechanism = client_sasl_mechanism
+        self.kafka.setup_interbroker_listener(client_protocol, use_separate_listener=False)
+        self.kafka.interbroker_sasl_mechanism = client_sasl_mechanism
+
+        self.kafka.start()
+
+        # create plaintext producer and consumer
+        self.create_producer_and_consumer()
+
+        # run produce/consume/validate loop while enabling a separate interbroker listener via rolling restart
+        self.run_produce_consume_validate(
+            self.roll_in_interbroker_listener, broker_protocol, broker_sasl_mechanism, True)
```

Comment:
Is there value in an upgrade test for this? Listener names are internal to the broker and as long as security protocols match, everything should just work. That is the reason why we didn't need to add upgrade tests when listener names were added all those releases ago. I think it is useful to have a test with listener names, but a single test, rather than all these combinations - there are six tests, each with rolling restarts in this file, adding a lot of time to the system tests.  I would suggest limiting to one combination for each test and without rolling restart. We could include this in `tests/kafkatest/tests/core/security_test.py` if it was just one test, but keeping it separate as it is now is fine too.

Reply:
@rajinisivaram I was going to argue that we do need to cover a use case where two listeners have the same security protocol since this is new functionality for this pr, but then this bit is covered in newly added `test_rolling_upgrade_phase_two_separate_interbroker_listener()` and two `test_rolling_upgrade_sasl_mechanism...` methods  in `security_rolling_upgrade_test.py` so I might get rid of this file altogether - I can just add a single case for removing a separate listener with rolling restart (I think there is some value in this, more for testing KafkaService itself, to ensure it closes the port correctly) and that's it. Wdyt?

Reply:
@rajinisivaram per your advice, reduced a number of tests to two and both live in security_rolling_upgrade_test now.